linux/git_mirror

```diff
@@ -289,15 +281,31 @@ function clone_submodules()
   # And the recursion goes on... foreach runs in sh, so I'm forcing bash. I need
   # to pass the prefix/displaypath variables from git into the called function,
   # so export them.
-  PREP_DIR="${PREP_DIR}" base_submodule_path="${base_submodule_path}" ${GIT} submodule foreach --quiet "export prefix displaypath; bash -euc 'unset GIT_DIR; source \"${GIT_MIRROR_SOURCE[0]}\"; clone_submodules'"
+  #
+  # REVIEW Why not simply loop over the submodules ourselves like
+  # just_git_functions:safe_git_submodule_update (which is called recursively
+  # via just_git_functions:_checkout_git_submodule) and now by
+  # just_git_airgap_repo.bsh:git_push_check_recursive_submodules. This would
+  # avoid the complications involved with exporting the required variables,
+  # keeping track of base_submodule_path, re-sourcing this file (and the
+  # info.env in update_submodules), etc. needed because git submodule foreach
+  # runs in an sh subshell
+  # RE We would first have to do a recursive clone, but we are effectively
+  # doing that here anyway, just manually
+  # NOTE We list submodules two ways in this function: at the beginning with
+  # get_submodule_names (which uses git config and works with init-only
+  # submodules) and, here, with git submodule foreach (which does not)
+  # NOTE We can not do this in update_submodules because we need to modify
+  # the remote URL after the submodule is init'd but before it is updated
```

Comment:
1. If it's not broke, don't fix it.
2. Isn't it better to let git do all the hard work rather than doing it all manually in bash?
3. ...